src/pipe/pipe.c

```diff
@@ -0,0 +1,132 @@
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <string.h>
+#include "pipe.h"
+#include "../shared/file_util.h"
+
+#define BUFFSZ 256
+
+int use_pipe(setting_t settings){
+    printf("usp %s\n", settings.password);
+    switch (settings.role) {
+        case SENDER:
+            return pipe_sender(settings);
+            break;
+        case RECEIVER:
+            return pipe_receiver(settings);
+            break;
+        default:
+            err_and_leave("Impossible role", 4);
+    }
+    return 0;
+}
+
+char* fifo_name(char* passwd){
+    printf("ffn %s\n", passwd);
+    char* myfifo = "/tmp/myfifo_";
+    // We have passwd
+    // We have myfifo
+    // new_string
+    // copy myfifo into new_string
+    // concatenate passwd to new_string
+    char* new_string = malloc(strlen(myfifo) + strlen(passwd) + 1);
+    strcpy(new_string, myfifo);
+    strcat(new_string, passwd);
+    return new_string;
+}
+
+void create_fifo(char* passwd){
+    printf("ffn %s\n", passwd);
+    char* new_string = fifo_name(passwd);
+    printf("%s\n", new_string);
+    int ret = mkfifo(new_string, 0666);
+    if (ret == 0){
+    } else if (ret == -1) {
+        switch (errno)
+        {
+        case EEXIST:
+            break;
+        default:
+        
+        printf("%d\n", errno);
+            err_and_leave("Failed to create pipe", 4);
+            break;
+        }
+    } else {
+        err_and_leave("", 4);
+    }
+}
+
+int pipe_sender(setting_t settings){
+    int fd_in, fd_fifo;
+    create_fifo(settings.password);
+    char* fifon = fifo_name(settings.password);
+    fd_fifo = open_file(fifon, O_WRONLY, 0);
```

Comment:
You could get the fifo_name() first and then use that to call create_fifo(). Or have the fifoname be a return value from create_fifo(). Now the program unnecessarily creates the name twice.

Reply:
This is a good point, @KocogluFerdi could you maybe modify the `create_fifo` function so that it returns the name used to create the fifo?

Reply:
I will fix it as soon as possible.